More rules for Electron

pachinko2821 opened the pull request:
Noticed a few rules could be added for Nodejs Electron, did just that.

- `electron_shell_openexternal`: Warns against possible malicious URL scheme use in `shell.openExternal` function.
- `electron_webview_allowpopups`: Warns against possible popup based attack vectors on webviews.
- `electron_browserview_loadurl`: Warns against possible malicious URL scheme use in `BrowserView.loadURL`.
- `electron_disabled_sandbox`: Warns against explicitly disabled sandboxing.

philipturnbull committed:
Only warn if string-concatenation is used

This is rule is false-positive heavy. Use taint-mode and only alert if the
argument to `sqlalchemy.text(...)` is built using string concatentation,
f-strings, etc.

philipturnbull committed:
Merge remote-tracking branch 'origin/develop' into improve-sqlalchemy-text-rule

philipturnbull committed:
Merge branch 'develop' into improve-sqlalchemy-text-rule

p4p3r committed:
Merge branch 'develop' into improve-sqlalchemy-text-rule

philipturnbull committed:
Merge branch 'develop' into improve-sqlalchemy-text-rule

pachinko2821 committed:
Adds rule for electron - shell.openExternal

pachinko2821 committed:
Updates rule for electron - shell.openExternal

pachinko2821 committed:
Adds typescript support

pachinko2821 committed:
Updated message for electron_shell_openExternal

pachinko2821 committed:
Add rule for electron - webview allowpopups

pachinko2821 committed:
Add rule for electron - dynamic browserview loadurl

pachinko2821 committed:
Add rule for electron - disabled sandbox

pachinko2821 committed:
Updates severity for rule - electron disabled sandbox

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/semgrep/semgrep-rules?pullRequest=3247) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/semgrep/semgrep-rules?pullRequest=3247) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/semgrep/semgrep-rules?pullRequest=3247) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/semgrep/semgrep-rules?pullRequest=3247) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/semgrep/semgrep-rules?pullRequest=3247) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/semgrep/semgrep-rules?pullRequest=3247) it.</sub>

p4p3r committed:
Add commutative_compop option to rule lints (#3248)

tartieret committed:
fix: Exclude access to request.user from view (#3241)

Co-authored-by: Claudio <claudio@r2c.dev>

LewisArdern committed:
Delete contrib directory (#3250)

This is not maintained by us, if people want to use most of these rules they can be found via https://semgrep.dev/p/nodejsscan

p4p3r committed:
Skip tests in CI for Pro only languages (#3251)

* Skip tests in CI for Pro only languages

* Skip tests in CI for Pro only languages

* Skip tests in CI for Pro only languages

p4p3r committed:
Add references to access-foreign-keys

inkz committed:
update django use-none-for-password-default rule (#3254)

* update django use-none-for-password-default rule

* Update use-none-for-password-default.py

* also add new testcase to the expected outcome of applying the autofix

---------

Co-authored-by: Pieter De Cremer <pieter@r2c.dev>

p4p3r committed:
Merge branch 'develop' into claudio/add-references

kurt-r2c committed:
Merge pull request #3255 from semgrep/claudio/add-references

Add references to access-foreign-keys

p4p3r committed:
Update disallow-old-tls-versions1.yaml (#3257)

* Update disallow-old-tls-versions1.yaml

* Update disallow-old-tls-versions1.java

---------

Co-authored-by: Lewis <LewisArdern@live.co.uk>

inkz committed:
update php tainted-sql-string rule (#3259)

0xDC0DE committed:
Update OSS Python cryptography rules (#3260)

0xDC0DE committed:
Update metadata for flask cookie and CSRF rules (#3261)

0xDC0DE committed:
Improve Flask-WTF CSRF rule (#3264)

* Improve Flask-WTF CSRF rules

* Add fixtest

0xDC0DE committed:
Add pattern-not for flask csrf tokens to django csrf rule (#3263)

normprovost committed:
update to rails unsafe reflection methods (#3265)

* update to rails unsafe reflection methods

* change from deep ellipsis to focus metavariable

---------

Co-authored-by: Kurt Boberg <98792107+kurt-r2c@users.noreply.github.com>

0xDC0DE committed:
Add fixes to python cryptography rules (#3267)

* Add fixes to python cryptography rules

* Move rule test syntaxt to reflect focus metavariable changes

* update mode recommendation to gcn

* update fixtest to reflect changes to fix

dependabot[bot] committed:
Bump jinja2 from 2.11.3 to 3.1.3 (#3271)

Bumps [jinja2](https://github.com/pallets/jinja) from 2.11.3 to 3.1.3.
- [Release notes](https://github.com/pallets/jinja/releases)
- [Changelog](https://github.com/pallets/jinja/blob/main/CHANGES.rst)
- [Commits](https://github.com/pallets/jinja/compare/2.11.3...3.1.3)

---
updated-dependencies:
- dependency-name: jinja2
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump tj-actions/changed-files from 23.1 to 41 in /.github/workflows (#3256)

Bumps [tj-actions/changed-files](https://github.com/tj-actions/changed-files) from 23.1 to 41.
- [Release notes](https://github.com/tj-actions/changed-files/releases)
- [Changelog](https://github.com/tj-actions/changed-files/blob/main/HISTORY.md)
- [Commits](https://github.com/tj-actions/changed-files/compare/v23.1...v41)

---
updated-dependencies:
- dependency-name: tj-actions/changed-files
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Pieter De Cremer (Semgrep) <pieter@r2c.dev>

Sjord committed:
Create rule for <meta https-equiv=...>

A fairly common mistake, possibly by a mass replacement of "http" by "https" throughout the source code.

hex0punk committed:
separate open redirect from tainted-url-host

p4p3r committed:
Remove redundant terraform rule

p4p3r committed:
Update message for AWS account id detection

p4p3r committed:
Merge pull request #3277 from semgrep/claudio/ebs

Remove redundant terraform rule

p4p3r committed:
Merge branch 'develop' into claudio/aws-account-id

armchairlinguist committed:
Update message to match version formats in rule

LewisArdern committed:
Merge pull request #3279 from semgrep/armchairlinguist/fix-tls-version-format

Update message to match version formats in rule

LewisArdern committed:
Merge branch 'develop' into claudio/aws-account-id

LewisArdern committed:
Merge branch 'develop' into improve-sqlalchemy-text-rule

LewisArdern committed:
Merge pull request #3046 from semgrep/improve-sqlalchemy-text-rule

Only warn if string-concatenation is used in `avoid-sqlalchemy-text`

LewisArdern committed:
Merge branch 'develop' into claudio/aws-account-id

LewisArdern committed:
Improve check-render rule

LewisArdern committed:
Merge pull request #3278 from semgrep/claudio/aws-account-id

Update message for AWS account id detection

LewisArdern committed:
Merge branch 'develop' into fix-ruby

kurt-r2c committed:
Merge pull request #3280 from semgrep/fix-ruby

Improve check-render rule

hex0punk committed:
removed open-redirect cases from tainted-url-host

hex0punk committed:
Merge branch 'develop' into separate-open-redirect

hex0punk committed:
update metedata for open-redirect

LewisArdern committed:
Merge branch 'develop' into https-equiv

LewisArdern committed:
Merge pull request #3273 from Sjord/https-equiv

Create rule for <meta https-equiv=...>

hex0punk committed:
Merge branch 'develop' into separate-open-redirect

LewisArdern committed:
Merge pull request #3275 from fullstorydev/separate-open-redirect

separate open redirect from tainted-url-host

philipturnbull committed:
Fix `yaml.semgrep.multi-line-message` CI error

philipturnbull committed:
Merge pull request #3284 from semgrep/https-equiv-message

Fix `html.correctness.https-equiv` description

kurt-r2c commented:
@pachinko2821 please resolve conflicts in your branch.

kurt-r2c committed:
remove non-literal header rule: fixed in PHP 5, PHP 4 EOL in 2008, PHP 5 EOL in 2019

kurt-r2c committed:
Merge pull request #3289 from semgrep/kb/remove-php-header-response-splitting

remove non-literal header rule:

brandonspark committed:
feat(autofix): fix fix indentation (#3288)

* fix

* another test

* fix some rules

* fix a bunch of rules hopefully

* redo

---------

Co-authored-by: Yoann Padioleau <pad@r2c.dev>

pachinko2821 committed:
Resolve merge conflict

pachinko2821 commented:
No clue what happend, I'll close this one and raise another PR
